tools/istio-iptables/pkg/validation/linux.go

```diff
@@ -0,0 +1,82 @@
+package validation
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"istio.io/istio/tools/istio-iptables/pkg/constants"
+	"net"
+	"syscall"
+)
+
+// <arpa/inet.h>
+func ntohs(n16 uint16) uint16 {
+	var bytes [2]byte
+	binary.BigEndian.PutUint16(bytes[:], n16)
+	return constants.HostEndian.Uint16(bytes[:])
+}
+
+// Recover the original address from redirect socket. Supposed to work for tcp over ipv4 and ipv6.
+func GetOriginalDestination(conn net.Conn) (daddr net.IP, dport uint16, err error) {
```

Comment:
Why do you need to recover original dest?
Just put a uuid in the request , can you explain the plan?

Reply:
OriginalDst is the proof of iptables setup. The advantage is that only tcp hand shake is required. Not tcp message is needed. The message is used to debug and it is best efforts